jupyterhub/apihandlers/services.py

```diff
@@ -31,8 +30,10 @@ def service_model(service):
 
 class ServiceListAPIHandler(APIHandler):
     @needs_scope('read:services')
-    def get(self):
+    def get(self, scope_filter=None):
         data = {name: service_model(service) for name, service in self.services.items()}
+        if scope_filter is not None:
+            data = dict(filter(lambda tup: tup[0] in scope_filter))
```

Comment:
I don't think this branch is tested, because this filter isn't valid (filter requires two arguments, the filter function and the thing being filtered).

What structure does  scope_filter have, exactly?

You can check for coverage of your PR (after tests finish) at https://app.codecov.io/gh/jupyterhub/jupyterhub/compare/3323/diff

Reply:
Thanks! The code coverage tool helps. And indeed, a scope_filter should be an iterable (set) of resource names. I'll write a test for it and fix it